src/reader/mod.rs

```diff
@@ -287,7 +287,7 @@ pub struct Reader<R> {
     /// reader
     reader: R,
     /// current buffer position, useful for debugging errors
-    buf_position: usize,
+    offset: usize,
```

Comment:
It would be helpful to clarify in the comments whether this is the offset is referring to the user-provided buffer, or the offset since the beginning of the file.

Reply:
Clarified